crates/server/src/admin/handlers/proposals.rs

```diff
@@ -0,0 +1,250 @@
+use std::sync::Arc;
+use std::vec;
+
+use axum::extract::Path;
+use axum::response::IntoResponse;
+use axum::{Extension, Json};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tower_sessions::Session;
+
+use crate::admin::service::ApiResponse;
+use crate::AdminState;
+
+//todo split it up into separate files
```

Comment:
Should be split into separate files, and most of the structs into the `primitives` crate

Reply:
Thats what todo says 😄